platform/state/platform_state.proto

```diff
@@ -0,0 +1,206 @@
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * The current state of platform consensus.<br/>
+ * This message stores the current consensus data for the platform
+ * in network state.
+ *
+ * The platform state SHALL represent the latest round's consensus.<br/>
+ * This data SHALL be used to ensure consistency and provide critical data for
+ * restart and reconnect.
+ */
+message PlatformState {
+    /**
+     * A consensus round.<br/>
+     * The round represented by this state.
+     * <p>
+     * This message SHALL represent the network state after handling of all
+     * transactions reaching consensus in all previous rounds.<br/>
+     * The first state (genesis state) SHALL have a round of 0.<br/>
+     * The first round to handle transactions SHALL be round `1`.
+     */
+    uint64 round = 1;
+
+    /**
+     * A consensus timestamp for this state.
+     * <p>
+     * This SHALL be the consensus timestamp of the first transaction in
+     * the current round.
+     */
+    proto.Timestamp consensus_timestamp = 2;
+
+    /**
+     * A version describing the current version of application software.
+     * <p>
+     * This SHALL be the software version that created this state.
+     */
+    proto.SemanticVersion creation_software_version = 3;
+
+    /**
+     * A number of non-ancient rounds.
+     * <p>
+     * This SHALL be the count of rounds considered non-ancient.
+     */
+    uint32 rounds_non_ancient = 4;
+
+    /**
+     * A snapshot of the consensus state at the end of the round.
+     * <p>
+     * This SHALL be used for restart/reconnect.
+     */
+    ConsensusSnapshot consensus_snapshot = 5;
+
+    /**
+     * A timestamp for the next scheduled time when a freeze will start.
+     * <p>
+     * If a freeze is not scheduled, this SHALL NOT be set.<br/>
+     * If a freeze is currently scheduled, this MUST be set, and MUST
+     * match the timestamp requested for that freeze.
+     */
+    proto.Timestamp freeze_time = 6;
+
+    /**
+     * A timestamp for the last time a freeze was performed.<br/>
+     * If not set, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 7;
+
+    /**
+     * A consensus node semantic version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * If birth round migration is complete, this SHALL be the _first_ software
+     * version that enabled birth round mode.
+     */
+    proto.SemanticVersion first_version_in_birth_round_mode = 8;
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 last_round_before_birth_round_mode = 9;
```

Comment:
Both of these fields are used only for birth round migration, and afterwards we will deprecate them. They should also have high field IDs.

Reply:
Fixed.